src/libraries/System.Private.CoreLib/src/System/IO/StringWriter.cs

```diff
@@ -151,22 +151,133 @@ public override void Write(string? value)
             }
 
-            _sb.Append(value);
+            return GetType() != typeof(StringWriter);
+        }
+
+        public override void Write(string format, params object?[] arg)
+        {
+            if (UseBase())
+            {
+                base.Write(format, arg: arg);
+            }
+            else
+            {
+                _sb.AppendFormat(format, args: arg);
+            }
         }
 
+        public override void Write(string format, object? arg0)
+        {
+            if (UseBase())
+            {
+                base.Write(format, arg0: arg0);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0);
+            }
+        }
+
+        public override void Write(string format, object? arg0, object? arg1)
+        {
+            if (UseBase())
+            {
+                base.Write(format, arg0: arg0, arg1: arg1);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0, arg1: arg1);
+            }
+        }
+
+        public override void Write(string format, object? arg0, object? arg1, object? arg2)
+        {
+            if (UseBase())
+            {
+                base.Write(format, arg0: arg0, arg1: arg1, arg2: arg2);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0, arg1: arg1, arg2: arg2);
+            }
+        }
+
+        public override void WriteLine(string format, params object?[] arg)
+        {
+            if (UseBase())
+            {
+                base.WriteLine(format, arg: arg);
+            }
+            else
+            {
+                _sb.AppendFormat(format, args: arg);
+                _sb.AppendLine();
+            }
+        }
+
+        public override void WriteLine(string format, object? arg0)
+        {
+            if (UseBase())
+            {
+                base.WriteLine(format, arg0: arg0);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0);
+                _sb.AppendLine();
+            }
+        }
+
+        public override void WriteLine(string format, object? arg0, object? arg1)
+        {
+            if (UseBase())
+            {
+                base.WriteLine(format, arg0: arg0, arg1: arg1);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0, arg1: arg1);
+                _sb.AppendLine();
+            }
+        }
+
+        public override void WriteLine(string format, object? arg0, object? arg1, object? arg2)
+        {
+            if (UseBase())
+            {
+                base.WriteLine(format, arg0: arg0, arg1: arg1, arg2: arg2);
+            }
+            else
+            {
+                _sb.AppendFormat(format, arg0: arg0, arg1: arg1, arg2: arg2);
+                _sb.AppendLine();
+            }
+        }
+
+        #endregion
```

Comment:
Are there tests covering these methods already?  If not, the PR should add some that covers both the base and derived cases.

The ref file also needs to be updated with the overrides:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs